### DISCUSSION TITLE
Project Structure

### DISCUSSION THREAD

**Discussion by Jsowens** [GTextreme169/SocialPlatform#2] [Ideas]
How is this project going to be structured? What sub projects do we need and what functionality to each contain?

**Comment by Jsowens** [comment]
Would the following work?

SocialPlatform
 - SocialPlatform.DesktopClient
 - SocialPlatform.WebClient
 - SocialPlatform.MobileClient
 - SocialPlatform.Core
 - SocialPlatform.Server

**Reply by GTextreme169** [reply]
Ideally Desktop and Mobile Clients would share most of their code. Additionally, Web, Desktop and Mobile should use the same services. All of the clients will share a library, and all projects would share core.

**Reply by JOwens-RTT** [reply]
Ok. So in addition to the projects above, we need a SocialPlatform.libClient and SocialPlatform.libServices? Then our project dependencies would look as such assuming we can completely roll Desktop and Mobile into one:

Social Platform
- SocialPlatform.Core
  - SocialPlatform.Server
  - SocialPlatform.libClient & SocialPlatform.libServices
    - SocialPlatform.Client
    - SocialPlatform.WebClient

**Reply by GTextreme169** [reply]
We can have both Libraries in the same project to simplify the project structure. (Ex. SocialPlatform.Client.Shared )
